app/cli.py

```diff
@@ -0,0 +1,239 @@
+
+    return clusters
+
+
+def get_pull_secret_data(pull_secret_file):
+    with open(pull_secret_file) as fd:
+        return json.load(fd)
+
+
+def get_base_install_config_data():
+    with open("app/manifests/install-config.yaml") as fd:
+        return yaml.safe_load(fd)
+
+
+def set_debug_os_flags():
+    os.environ["OCM_PYTHON_WRAPPER_LOG_LEVEL"] = "DEBUG"
+    os.environ["OPENSHIFT_PYTHON_WRAPPER_LOG_LEVEL"] = "DEBUG"
+
+
+def get_local_ssh_key():
+    with open(os.path.expanduser("~/.ssh/id_rsa.pub")) as fd:
+        return fd.read().strip()
+
+
+@click.command()
+@click.option("-i", "--install", is_flag=True, help="Install Openshift cluster/s")
+@click.option("-u", "--uninstall", is_flag=True, help="Uninstall Openshift cluster/s")
+@click.option(
+    "-p",
+    "--parallel",
+    help="Run clusters install/uninstall in parallel",
+    is_flag=True,
+    show_default=True,
+)
+@click.option(
+    "--clusters-install-data-directory",
+    help="""
+\b
+Path to cluster install data
+    For install this will be used to store the install data.
+    For uninstall this will be used to uninstall the cluster and must be provided.
+""",
+    default=os.environ.get(
+        "CLUSTER_INSTALL_DATA_DIRECTORY",
+        "/openshift-cli-installer/clusters-install-data",
+    ),
+    type=click.Path(exists=True),
+    show_default=True,
+)
+@click.option(
+    "--pull-secret-file",
+    help="Path to pull secret json file, can be obtained from console.redhat.com",
+    required=True,
+    default=os.environ.get("PULL_SECRET"),
+    type=click.Path(exists=True),
+    show_default=True,
+)
+@click.option(
+    "-c",
+    "--cluster",
+    type=DictParamType(),
+    help="""
+\b
+Cluster/s to install.
+Format to pass is:
+    'name=cluster1;baseDomain=aws.domain.com;platform=aws;region=us-east-2;version=4.14.0-ec.2'
+Required parameters:
+    name: Cluster name.
+    baseDomain: Base domain for the cluster.
+    platform: Cloud platform to install the cluster on. (Currently only AWS supported).
+    region: Region to use for the cloud platform.
+    version: Openshift cluster version to install
+
+Every parameter in install-config.yaml can be override by the user.
+For example:
+    fips=true
+    compute_platform_aws_type=m5.xlarge
+    """,
+    required=True,
+    multiple=True,
+)
+@click.option("--debug", help="Enable debug logs", is_flag=True)
+def main(
+    install,
+    uninstall,
+    pull_secret_file,
+    parallel,
+    cluster,
+    debug,
+    clusters_install_data_directory,
+):
+    """
+    Install/Uninstall Openshift cluster/s
+    """
+    set_and_verify_aws_credentials()
+    if not (install or uninstall):
+        raise ValueError("One of install/uninstall must be specified")
+
+    if debug:
+        set_debug_os_flags()
+
+    base_install_config = update_base_install_config(pull_secret_file=pull_secret_file)
+    clusters = generate_cluster_dir_path(
+        clusters=cluster, base_directory=clusters_install_data_directory
+    )
```

Comment:
for uninstall we assume we have the proper file in place; we can call `update_base_install_config` from `create_install_config_file`?

Reply:
For uninstall we use the first call of `generate_cluster_dir_path`, I don't understand what you mean here